#!/usr/bin/env python
PACKAGE = "asr_fake_object_recognition"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("frustum_mode", int_t, 0, "Specifies which camera frustum is used for the pose rating of the object (0=both needed, 1=only one needed, 2=use the left one, 3=use the right one)", 0, 0, 3)

gen.add("use_camera_pose", bool_t, 0, "If false all objects in the config file are published without checking their pose relative to the camera", True)

gen.add("use_pose_invalidation", bool_t, 0, "If true randomly invalidates found poses based on prob_pose_invalidation", True)
gen.add("prob_pose_invalidation", double_t, 0, "The probability which is used to invalidate poses", 0.1, 0, 1)

#Position noise variables
gen.add("use_position_noise", bool_t, 0, "If true randomly alters the positions of the found objects", True)
gen.add("pos_noise_normal_dist_mean", double_t, 0, "The mean value of the normal distribution used for the generation of position errors", 0.0, -10.0, 10.0)
gen.add("pos_noise_normal_dist_dev", double_t, 0, "The standard deviation value of the normal distribution used for the generation of position errors", 0.01, 0.0001, 1.0)

#Orientation noise variables
gen.add("use_orientation_noise", bool_t, 0, "If true randomly alters the orientations of the found objects", True)
gen.add("or_x_noise_normal_dist_mean", double_t, 0, "The mean value of the normal distribution used for the generation of orientation errors (x-axis)", 0.0, -10.0, 10.0)
gen.add("or_x_noise_normal_dist_dev", double_t, 0, "The standard deviation value of the normal distribution used for the generation of orientation errors (x-axis)", 0.02, 0.0001, 3.14)
gen.add("or_y_noise_normal_dist_mean", double_t, 0, "The mean value of the normal distribution used for the generation of orientation errors (y-axis)", 0.0, -10.0, 10.0)
gen.add("or_y_noise_normal_dist_dev", double_t, 0, "The standard deviation value of the normal distribution used for the generation of orientation errors (y-axis)", 0.02, 0.0001, 3.14)
gen.add("or_z_noise_normal_dist_mean", double_t, 0, "The mean value of the normal distribution used for the generation of orientation errors (z-axis)", 0.0, -10.0, 10.0)
gen.add("or_z_noise_normal_dist_dev", double_t, 0, "The standard deviation value of the normal distribution used for the generation of orientation errors (z-axis)", 0.02, 0.0001, 3.14)

gen.add("config_file_path", str_t, 0, "The path to the xml-file containing the scene objects", "./config/objects.xml")

exit(gen.generate(PACKAGE, "asr_fake_object_recognition", "FakeObjectRecognition"))
